validator/lib/jsonapi/validator/resource.rb

```diff
@@ -0,0 +1,123 @@
+require 'jsonapi/validator/document'
+require 'jsonapi/validator/relationship'
+
+module JSONAPI
+  module Validator
+    class Resource
```

Comment:
How about the following:
1. The current validator, without the whitelisting/type checking becomes what is currently called the parser;
2. validator becomes an external gem.